layers/+spacemacs/spacemacs-layouts/funcs.el

```diff
@@ -608,3 +608,16 @@ FRAME defaults to the current frame."
                                         (eyebrowse--get 'last-slot frame))
                                   (get-frame-persp frame)
                                   frame))
+
+;; shell-pop integration
+
+(defun spacemacs//set-persp-specific-shell (arg)
+  "To be called after persp-switch. Make the default shell number to use equal to the layout number."
+  (progn
+    (loop for persp in (persp-names-current-frame-fast-ordered) and i from 1
+          when (string= persp arg)
+          do (progn (setq shell-pop-last-shell-buffer-index i) (return)))))
```

Comment:
No. It doesn't work. The shell popped will be the current layout number. I don't know enough about emacs lisp to create a map and get/set values. But changing the `make-shell-pop-command` seems like a better idea. 

Btw, inside `make-shell-pop-command`, there is an `if` that says no popup if index equals 4 which I do not get.